geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SftArgResolver.scala

```diff
@@ -22,84 +23,100 @@ import scala.util.{Failure, Success, Try}
+             || s.contains("geomesa.sfts")  => CONFSTR
+      case s if s.matches(specStrReg)       => SPECSTR
+      case s if s.matches(fileNameReg)
+             || s.contains("/")             => PATH
+      case s if s.matches(specStrRegWError) => SPECSTR
+      case _                                => NAME
+    }
+  }
+
+  override def parseOption(args: argTuple): Option[SimpleFeatureType] = getLoadedSft(args)
+
+  override val parseMethodList = List[argTuple => resEither](
+    parseSpecString,
+    parseConfStr,
+    parseSpecStringFile,
+    parseConfFile
+  )
 
   // gets an sft from simple feature type providers on the classpath
-  private[SftArgResolver] def getLoadedSft(specArg: String, name: String): Option[SimpleFeatureType] = {
-    SimpleFeatureTypeLoader.sfts.find(_.getTypeName == specArg).map { sft =>
-      if (name == null || name == sft.getTypeName) sft else SimpleFeatureTypes.renameSft(sft, name)
+  private[SftArgResolver] def getLoadedSft(args: argTuple): Option[SimpleFeatureType] = {
+    SimpleFeatureTypeLoader.sfts.find(_.getTypeName == args._1).map { sft =>
+      if (args._2 == null || args._2 == sft.getTypeName) sft else SimpleFeatureTypes.renameSft(sft, args._2)
     }
   }
 
   // gets an sft based on a spec string
-  private[SftArgResolver] def parseSpecString(specArg: String, name: String): Option[SimpleFeatureType] =
-    Option(name).flatMap { featureName =>
-      Try(SimpleFeatureTypes.createType(featureName, specArg)) match {
-        case Success(sft) => Some(sft)
-        case Failure(e) =>
-          logger.debug(s"Unable to parse sft spec from string $specArg with error ${e.getMessage}")
-          None
-      }
+  private[SftArgResolver] def parseSpecString(args: argTuple): resEither = {
+    Try(SimpleFeatureTypes.createType(
+        Option(args._2).getOrElse(throw new Throwable("Null feature name provided. " +
+          "Unable to parse spec string from null value.")),
+      args._1)
+    ) match {
+      case Success(sft) => Right(sft)
+      case Failure(e) => Left((s"Unable to parse sft spec from string ${args._1}.", e, SPECSTR))
     }
+  }
 
   // gets an sft based on a spec string
-  private[SftArgResolver] def parseSpecStringFile(specArg: String, name: String): Option[SimpleFeatureType] =
-    Option(specArg).map(new File(_)).flatMap { file =>
-      Try(SimpleFeatureTypes.createType (name, FileUtils.readFileToString(file))) match {
-        case Success(sft) => Some(sft)
-        case Failure(e) =>
-          logger.debug(s"Unable to parse sft spec from string $specArg with error ${e.getMessage}")
-          None
-      }
+  private[SftArgResolver] def parseSpecStringFile(args: argTuple): resEither =
+    Try(SimpleFeatureTypes.createType(
+      Option(args._2).getOrElse(throw new Throwable("Null feature name provided. " +
+        "Unable to parse spec string from null value.")),
+      FileUtils.readFileToString(new File(Option(args._1).getOrElse(""))))
```

Comment:
Does it make sense to replace the `getOrElse("")` with an exception like 'No input file specified' ?